### PR TITLE
Update Cuda provider config API

### PR DESCRIPTION
**Pull request by vinser52**
### Description
Change the Cuda provider config as we agreed in #844

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly
- [x] New tests added, especially if they will fail without my changes
- [x] Added/extended example(s) to cover this functionality

### CODE REVIEW

File: test/providers/ipc_cuda_prov_consumer.c [hunk 1]
```diff
@@ -22,23 +22,76 @@ int main(int argc, char *argv[]) {
...
+    ret = create_context(hDevice, &hContext);
+    if (ret != 0) {
+        fprintf(stderr, "create_context() failed!\n");
+        return -1;
+    }
+
+    umf_cuda_memory_provider_params_handle_t cu_params = NULL;
+    umf_result_t umf_result = umfCudaMemoryProviderParamsCreate(&cu_params);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "Failed to create level zero params!\n");
```
Comment (kswiecicki) [inline test/providers/ipc_cuda_prov_consumer.c:43]: `[...] CUDA memory provider params!`, same with the rest of the error messages in the consumer and producer.

**Review by kswiecicki — APPROVED** [review]

File: test/providers/ipc_cuda_prov_consumer.c (hunk 1 above)
Comment (vinser52) [reply test/providers/ipc_cuda_prov_consumer.c]: Done

File: include/umf/providers/provider_cuda.h [hunk 2]
```diff
@@ -14,12 +14,44 @@
...
-} cuda_memory_provider_params_t;
+struct umf_cuda_memory_provider_params_t;
+
+typedef struct umf_cuda_memory_provider_params_t
+    *umf_cuda_memory_provider_params_handle_t;
+
+/// @brief  Create a struct to store parameters of the Cuda memory provider.
+/// @param  hParams [out] handle to the newly created parameters struct.
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfCudaMemoryProviderParamsCreate(
```
Comment (bratpiorka) [inline include/umf/providers/provider_cuda.h:25]: please always write "CUDA" using capital letters as this is an acronym

**Comment by bratpiorka** [comment]
general note: when referring to a specific UMF pool provider/implementation, we should always use capital letters, e.g. we should write ‘UMF CUDA Memory Provider is an example of a memory provider implemented in UMF’ etc. We should also never write ‘Cuda’. Always use ‘CUDA’ in comments, and ‘cuda’ only as part of C variables/names, such as ‘umf_cuda_params’ etc.

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: test/providers/ipc_cuda_prov_consumer.c [hunk 3]
```diff
@@ -22,23 +22,75 @@ int main(int argc, char *argv[]) {
...
+        goto destroy_provider_params;
+    }
 
     umf_disjoint_pool_params_handle_t pool_params = NULL;
 
-    umf_result_t umf_result = umfDisjointPoolParamsCreate(&pool_params);
+    umf_result = umfDisjointPoolParamsCreate(&pool_params);
     if (umf_result != UMF_RESULT_SUCCESS) {
         fprintf(stderr, "Failed to create pool params!\n");
         return -1;
```
[the remark this answers is not in the document]
Comment (vinser52) [reply test/providers/ipc_cuda_prov_consumer.c]: Done

File: test/providers/ipc_cuda_prov_producer.c [hunk 4]
```diff
@@ -22,23 +22,75 @@ int main(int argc, char *argv[]) {
...
+        goto destroy_provider_params;
+    }
 
     umf_disjoint_pool_params_handle_t pool_params = NULL;
 
-    umf_result_t umf_result = umfDisjointPoolParamsCreate(&pool_params);
+    umf_result = umfDisjointPoolParamsCreate(&pool_params);
     if (umf_result != UMF_RESULT_SUCCESS) {
         fprintf(stderr, "Failed to create pool params!\n");
         return -1;
```
[the remark this answers is not in the document]
Comment (vinser52) [reply test/providers/ipc_cuda_prov_producer.c]: Done

File: include/umf/providers/provider_cuda.h (hunk 2 above)
Comment (vinser52) [reply include/umf/providers/provider_cuda.h]: Done

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**